banyand/stream/query.go

```diff
@@ -18,27 +18,26 @@
 package stream
 
 import (
+	"bytes"
 	"container/heap"
 	"context"
 	"errors"
 	"fmt"
 	"sort"
 
-	"go.uber.org/multierr"
-
 	"github.com/apache/skywalking-banyandb/api/common"
 	databasev1 "github.com/apache/skywalking-banyandb/api/proto/banyandb/database/v1"
 	modelv1 "github.com/apache/skywalking-banyandb/api/proto/banyandb/model/v1"
 	"github.com/apache/skywalking-banyandb/banyand/internal/storage"
 	"github.com/apache/skywalking-banyandb/pkg/convert"
-	itersort "github.com/apache/skywalking-banyandb/pkg/iter/sort"
 	"github.com/apache/skywalking-banyandb/pkg/logger"
 	"github.com/apache/skywalking-banyandb/pkg/partition"
 	pbv1 "github.com/apache/skywalking-banyandb/pkg/pb/v1"
 )
 
 type queryOptions struct {
-	elementRefMap map[common.SeriesID][]int64
+	filteredRefMap map[common.SeriesID][]int64
```

Comment:
Use `posting.List` instead of `[]int64`. It provides a more compact memory footprint.